lib/data.dart

```diff
@@ -118,21 +117,50 @@ Future<TA> getTaById(String taId) {
 Future<Course> getCourseById(String courseId) {
   return db.collection('courses').doc(courseId).get().then((courseDoc) {
     var courseData = courseDoc.data();
-    var courseTas = courseTaIds(courseData);
-    return new Course(
-        courseData['name'], courseDoc.id, courseData['yearId'], courseTas);
+    return new Course(courseData['name'], courseDoc.id, courseData['yearId']);
   });
 }
 
-Future<TaCourse> getTaCoursePair(String docId) {
-  return db.collection('ta-course').doc(docId).get().then((doc) {
+Future<TaCourse> getTaCoursePair(String taId, String courseId) {
+  return db
+      .collection('ta-course')
+      .where('taId', isEqualTo: taId)
+      .where('courseId', isEqualTo: courseId)
+      .get()
+      .then((snap) {
+    var doc = snap.docs.first;
     if (doc.exists == false) {
       return null;
     }
 
     var docData = doc.data();
+    double rating = 0;
+    if (docData['rating'] != null) rating = (docData['rating']) * 1.0;
```

Comment:
why * 1.0 is needed? kinda toDouble?)

Reply:
Yes, exactly. I did not realize that double.parse() exists.